substrate/frame/opf/src/lib.rs

```diff
@@ -0,0 +1,662 @@
+		/// At the end of the voting period, rewards are calculated based on the total user amount
+		/// attributed to each project by the user’s votes.
+		///
+		/// ### Parameters
+		/// - `project_id`: The account that will receive the reward.
+		/// - `amount`: Amount that will be locked in user’s balance to nominate a project.
+		/// - `fund`: Parameter that defines if user’s vote is in favor (*true*), or against
+		///   (*false*)
+		/// the project funding.
+		/// - `conviction`: Used to calculate the value allocated to the project, & determine
+		/// when the voter's funds will be unlocked. Amount actually locked is the amount without
+		/// conviction  
+		///
+		/// ### Errors
+		/// - [`Error::<T>::NotEnoughFunds`]: The user does not have enough balance to cast a vote
+		///  
+		/// ## Events
+		/// - [`Event::<T>::VoteCasted { who, project_id }`]: User's vote successfully submitted
+		#[pallet::call_index(3)]
+		#[pallet::weight(<T as Config>::WeightInfo::vote(T::MaxProjects::get()))]
+		pub fn vote(
+			origin: OriginFor<T>,
+			project_id: ProjectId<T>,
+			#[pallet::compact] amount: BalanceOf<T>,
+			fund: bool,
+			conviction: Conviction,
+		) -> DispatchResult {
+			let voter = ensure_signed(origin.clone())?;
+			// Get current voting round & check if we are in voting period or not
+			Self::period_check()?;
+
+			let infos = WhiteListedProjectAccounts::<T>::get(&project_id)
+				.ok_or(Error::<T>::NoProjectAvailable)?;
+			let ref_index = infos.index;
+
+			// Funds lock is handled by the opf pallet
+			let conv = Conviction::None;
+			let u128_amount = amount.saturated_into::<u128>();
+			let converted_amount =
+				T::Conviction::u128_to_balance(u128_amount).ok_or("Failed Conversion!!!")?;
+			let account_vote = T::Conviction::vote_data(fund, conv, converted_amount);
+			if Votes::<T>::contains_key(&ref_index, &voter) {
+				T::Conviction::try_remove_vote(&voter, ref_index.into())?;
+			}
+			T::Conviction::try_vote(&voter, ref_index.into(), account_vote)?;
+
+			Self::try_vote(voter.clone(), project_id.clone(), amount, fund, conviction)?;
+
+			Self::deposit_event(Event::<T>::VoteCasted { who: voter, project_id });
+
+			Ok(())
+		}
+
+		/// OPF vote removal logic
+		///
+		/// ## Dispatch Origin
+		///
+		/// Must be signed
+		///
+		/// ## Details
+		///
+		/// This extrinsic allows users to remove a casted vote, as long as it is within the
+		/// vote-casting period.
+		///
+		/// ### Parameters
+		/// - `project_id`: The account that will receive the reward.
+		///
+		/// ### Errors
+		/// - [`Error::<T>::NoProjectAvailable`]: No project found under this project_id
+		///  
+		/// ## Events
+		/// - [`Event::<T>::VoteRemoved { who, project_id }`]: User's vote successfully removed
+		#[pallet::call_index(4)]
+		#[pallet::weight(<T as Config>::WeightInfo::remove_vote(T::MaxProjects::get()))]
+		pub fn remove_vote(origin: OriginFor<T>, project_id: ProjectId<T>) -> DispatchResult {
```

Comment:
for the same reason this call should not exist either.

Reply:
see previoous comment